src/Engine/Trigger/TriggerRepeat.cpp

```diff
@@ -0,0 +1,218 @@
+    triggered = true;
+}
+
+std::vector<EventStruct> *TriggerRepeat::GetEventVector() {
+    return &eventVector;
+}
+
+Entity* TriggerRepeat::GetOwningEntity() {
+    return owningEntity;
+}
+
+void TriggerRepeat::SetOwningEntity(Entity* value) {
+    owningEntity = value;
+}
+
+void TriggerRepeat::InitTriggerUID() {
+
+    if (GetEntity(targetEntityUID) != nullptr)
+        targetEntity.push_back(GetEntity(targetEntityUID));
+    if (GetEntity(collidedEntityUID) != nullptr)
+        collidedEntity.push_back(GetEntity(collidedEntityUID));
+    if (GetEntity(owningEntityUID) != nullptr)
+        owningEntity = GetEntity(owningEntityUID);
+
+}
+
+void TriggerRepeat::Process() {
+    // ADD SUPPORT FOR VECTOR
+    if (triggered) {
+        Managers().scriptManager->ExecuteScriptMethod(targetEntity.front(), targetFunction.front());
+        triggered = false;
+    }
+}
+
+void TriggerRepeat::Update() {
+
+    if (owningEntity != nullptr) {
+        Managers().physicsManager->SetPosition(triggerVolume, owningEntity->GetWorldPosition());
+
+        if (owningEntity->GetComponent<Component::Shape>() != nullptr) {
+            Managers().physicsManager->SetShape(triggerVolume, owningEntity->GetComponent<Component::Shape>()->GetShape());
+        }
+    }
+}
+
+Json::Value TriggerRepeat::Save() {
+    Json::Value component;
+
+    component["trigger"] = "A Trigger";
+    component["triggerName"] = name;
+    component["triggerActive"] = startActive;
+    component["triggerDelay"] = delay;
+    component["triggerCooldown"] = cooldown;
+    component["triggerCharges"] = triggerCharges;
+
+    if (!targetFunction.empty())
+        component["triggerFunction"] = targetFunction.front(); // ADD SUPPORT FOR VECTOR
+    if (!targetEntity.empty())
+        component["triggerTargetEntity"] = targetEntity.front()->GetUniqueIdentifier(); // ADD SUPPORT FOR VECTOR
+    if (!collidedEntity.empty())
+        component["triggerCollidedEntity"] = collidedEntity.front()->GetUniqueIdentifier(); // ADD SUPPORT FOR VECTOR
+
+    if (!eventVector.empty()) {
+        component["triggerEventStruct_EventID"] = eventVector.front().m_eventID; // ADD SUPPORT FOR VECTOR
+        component["triggerEventStruct_ShapeID"] = eventVector.front().m_shapeID; // ADD SUPPORT FOR VECTOR
+        component["triggerEventStruct_TargetID"] = eventVector.front().m_targetID; // ADD SUPPORT FOR VECTOR
+        component["triggerEventStruct_ScriptID"] = eventVector.front().m_scriptID; // ADD SUPPORT FOR VECTOR
+        component["triggerEventStruct_Check_0"] = eventVector.front().check[0]; // ADD SUPPORT FOR VECTOR
+        component["triggerEventStruct_Check_1"] = eventVector.front().check[1]; // ADD SUPPORT FOR VECTOR
+        component["triggerEventStruct_Check_2"] = eventVector.front().check[2]; // ADD SUPPORT FOR VECTOR
+        component["triggerEventStruct_Check_3"] = eventVector.front().check[3]; // ADD SUPPORT FOR VECTOR
+    }
+
+    if (GetEventVector()->front().check[0],
```

Comment:
Commas should be and comparison because otherwise the value is discarded due to comma and there are no side effects we would otherwise be interested in. The same applies to the other branches.